### PR TITLE
[Refactoring] Move tests from Junit 4 to Junit 5 in mailbox-cassandra module

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/AbstractSubscriptionManagerTest.java [hunk 1]
```diff
@@ -38,7 +39,8 @@
 
     protected abstract SubscriptionManager createSubscriptionManager();
 
-    protected void setup() {
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/AbstractSubscriptionManagerTest.java]: This class should become an interface?

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerQuotaTest.java [hunk 2]
```diff
@@ -67,8 +63,8 @@
     
     protected abstract QuotaManager createQuotaManager(MaxQuotaManager maxQuotaManager, CurrentQuotaManager currentQuotaManager);
 
-    @Before
-    public void setUp() throws Exception {
+    @BeforeEach
+    void setUp() throws Exception {
```
Comment (chibenwa) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerQuotaTest.java]: This class should be an interface?

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerStressTest.java [hunk 3]
```diff
@@ -53,8 +54,9 @@
     protected abstract T provideManager() throws MailboxException;
 
     protected abstract EventBus retrieveEventBus(T mailboxManager);
-    
-    protected void setUp() throws Exception {
+
+    @BeforeEach
+    void setUp() throws Exception {
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerStressTest.java]: This class should become an interface?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraSubscriptionManagerTest.java [hunk 4]
```diff
@@ -61,11 +60,6 @@
     @RegisterExtension
     static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraSubscriptionModule.MODULE);
 
-    @BeforeEach
-    void init() {
-        super.setup();
-    }
-
     @AfterEach
     void close() throws SubscriptionException {
```
Comment (chibenwa) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraSubscriptionManagerTest.java]: Is this still required? Can't the parent class do the @AfterEach?

**Comment by Arsnael** [comment]
> We can consider doing the `abstract class` -> `interface contract` migration later, I perfectly agree to get this done in upcoming pull requests.

I agree, it's big enough, but good point, will try to do this work in an other PR

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**